src/cirrus/management/deployment_pointer.py

```diff
@@ -97,68 +36,115 @@ class Pointer:
     pointer: Pointer
+    """
+    Class to retrieve deployment pointer and using pointer
+    for building deployment
+    """
 
     @classmethod
     def _from_parameter(
         cls,
         parameter: dict[str, Any],
-        prefix: str = "",
+        name: str = "",
     ) -> DeploymentPointer:
-        name = parameter["Name"][len(prefix) :]
         return cls(
             name=name,
             pointer=Pointer.from_string(parameter["Value"]),
-            prefix=prefix,
         )
 
     @classmethod
-    def get(
+    def _get_parameters_by_path(cls, ssm, prefix: str, next: str = ""):
+        return ssm.get_parameters_by_path(
+            Path=prefix,
+            Recursive=True,
+            NextToken=next,
+        )
+
+    @classmethod
+    def _get_parameters(
+        cls,
+        prefix: str,
+        session: boto3.Session,
+    ) -> list[dict[str, Any]]:
+        ssm = get_client("ssm", region=session.region_name, session=session)
+        resp = cls._get_parameters_by_path(ssm, prefix)
+        parameters = resp["Parameters"]
+
+        # handle pagination
+        while "NextToken" in resp:
+            resp = cls._get_parameters_by_path(ssm, prefix, resp["NextToken"])
+            parameters = parameters + resp["Parameters"]
+
+        return parameters
+
+    @classmethod
+    def get_pointer(
         cls,
         deployment_name: str,
-        deployment_prefix: str = DEFAULT_CIRRUS_DEPLOYMENT_PREFIX,
         region: str | None = None,
         session: boto3.Session | None = None,
     ):
+        """Get pointer to deployment in param store.  Pointer is one param"""
         ssm = get_client("ssm", region=region, session=session)
         return cls._from_parameter(
-            ssm.get_parameter(
-                Name=f"{deployment_prefix}{deployment_name}",
-            )["Parameter"],
-            prefix=deployment_prefix,
+            ssm.get_parameter(Name=f"{DEPLOYMENTS_PREFIX}{deployment_name}")[
+                "Parameter"
+            ],
+            name=deployment_name,
         )
 
     @classmethod
-    def list(
+    def list_deployments(
         cls,
-        deployment_prefix: str = DEFAULT_CIRRUS_DEPLOYMENT_PREFIX,
-        region: str | None = None,
-        session: boto3.Session | None = None,
-    ) -> Iterator[DeploymentPointer]:
-        ssm = get_client("ssm", region=region, session=session)
-        resp = ssm.get_parameters_by_path(Path=deployment_prefix)
-        for param in resp["Parameters"]:
-            yield cls._from_parameter(
-                param,
-                prefix=deployment_prefix,
-            )
-
-    def get_config(
+        session: boto3.Session,
+    ) -> list[str]:
+        """Retrieve and list names of available deployments in parameter
+        store"""
+        parameters = cls._get_parameters(
+            DEPLOYMENTS_PREFIX,
+            session=session,
+        )
+
+        return [param["Name"].split(DEPLOYMENTS_PREFIX)[1] for param in parameters]
```

Comment:
With the generator version of `get_parameters_by_path` I am pretty sure this can be written as:

```suggestion
    ) -> Iterator[str]:
        """Retrieve and list names of available deployments in parameter
        store"""
        yield from (
            param["Name"].removeprefix(DEPLOYMENTS_PREFIX)
            for param in get_parameters_by_path(DEPLOYMENTS_PREFIX, session=session)
        )
```

Doing so convert it to a generator as well, so call sites might also need a slight update.

Reply:
I think I'm starting to get a feel for `yield`, I really like that you can avoid an intermediate list assignment like this, the efficency/memory improvement is obvious.